File: scripts/lib/resolve-ids.ts
import type { SupabaseClient } from '@supabase/supabase-js'

// Resolves a batch of lis_member_ids to bioguide_ids using the legislators table.
// Returns a map of lis_id → bioguide_id for fast lookup.
export async function buildLisMap(
  supabase: SupabaseClient,
  lisIds: string[]
): Promise<Map<string, string>> {
  if (lisIds.length === 0) return new Map()

  const { data, error } = await supabase
    .from('legislators')
    .select('lis_id, bioguide_id')
    .in('lis_id', lisIds)

  if (error) throw new Error(`buildLisMap: ${error.message}`)

  const map = new Map<string, string>()
  for (const row of data ?? []) {
    if (row.lis_id && row.bioguide_id) map.set(row.lis_id, row.bioguide_id)
  }
  return map
}

File: pipeline/lib/resolve_ids.py
"""
Resolves LIS member IDs to bioguide IDs using the legislators table.
Port of scripts/lib/resolve-ids.ts
"""
from supabase import Client


def build_lis_map(supabase: Client, lis_ids: list[str]) -> dict[str, str]:
    """Returns a dict of lis_id → bioguide_id for the given lis_ids."""
    if not lis_ids:
        return {}

    resp = (
        supabase.table("legislators")
        .select("lis_id, bioguide_id")
        .in_("lis_id", lis_ids)
        .execute()
    )

    result: dict[str, str] = {}
    for row in resp.data or []:
        if row.get("lis_id") and row.get("bioguide_id"):
            result[row["lis_id"]] = row["bioguide_id"]
    return result
